components/conversations/EmojiReactionPicker.js

```diff
@@ -58,34 +72,43 @@ const ReactionButton = styled(StyledRoundButton).attrs({ isBorderless: true, but
     `}
 `;
 
-const getOptimisticResponse = (comment, emoji, isAdding) => {
-  const userReactions = comment.userReactions || [];
+const getOptimisticResponse = (entity, emoji, isAdding) => {
+  const userReactions = entity.userReactions || [];
+  const { __typename } = entity;
+  const fieldName = __typename === 'Update' ? 'update' : 'comment';
   if (isAdding) {
-    const newCount = (comment.reactions[emoji] || 0) + 1;
+    const newCount = (entity.reactions[emoji] || 0) + 1;
+
     return {
       __typename: 'Mutation',
-      addCommentReaction: {
-        __typename: 'Comment',
-        id: comment.id,
-        reactions: { ...comment.reactions, [emoji]: newCount },
-        userReactions: [...userReactions, emoji],
+      addEmojiReaction: {
+        __typename: 'addEmojiReaction',
+        [fieldName]: {
+          __typename,
+          id: entity.id,
+          reactions: { ...entity.reactions, [emoji]: newCount },
+          userReactions: [...userReactions, emoji],
+        },
       },
     };
   } else {
-    const newCount = (comment.reactions[emoji] || 0) - 1;
-    const reactions = { ...comment.reactions, [emoji]: newCount };
+    const newCount = (entity.reactions[emoji] || 0) - 1;
+    const reactions = { ...entity.reactions, [emoji]: newCount };
 
     if (!reactions[emoji]) {
       delete reactions[emoji];
     }
 
     return {
       __typename: 'Mutation',
-      removeCommentReaction: {
-        __typename: 'Comment',
-        id: comment.id,
-        reactions,
-        userReactions: userReactions.filter(userEmoji => userEmoji !== emoji),
+      removeEmojiReaction: {
+        __typename: 'removeEmojiReaction',
```

Comment:
To be confirmed by looking at the actual mutation response in the Network logs, but I think we want:

```suggestion
        __typename: 'EmojiReactionsResponse',
```

Reply:
@Betree: Same as above, I think it should be, `EmojiReactionOutput`. 😄

Reply:
@Betree : The point here is to match the optimistic response with the actual response right? 🤔

Reply:
@SudharakaP It is, but haven't we renamed `EmojiReactionOutput` to `EmojiReactionResponse`?

Reply:
@Betree : Ah yes, I didn't see your API review until just now. I've renamed it both here and there. 😄